test/test_examples_2d_p4est.jl

```diff
@@ -15,12 +15,30 @@ EXAMPLES_DIR = joinpath(pathof(Trixi) |> dirname |> dirname, "examples", "2d")
       linf = [6.437440532947036e-5])
   end
 
-  @testset "elixir_advection_p4est_non_conforming.jl" begin
-    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_p4est_non_conforming.jl"),
+  @testset "elixir_advection_p4est_non_conforming_flag.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_p4est_non_conforming_flag.jl"),
       l2   = [4.634288969205318e-4],
       linf = [4.740692055057893e-3])
   end
 
+  @testset "elixir_advection_p4est_non_conforming_flag_unstructured.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_p4est_non_conforming_flag_unstructured.jl"),
+      l2   = [3.038384623519386e-3],
+      linf = [6.324792487776842e-2])
+  end
+
+  @testset "elixir_advection_amr_solution_independent_p4est.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_amr_solution_independent_p4est.jl"),
+      l2   = [7.945073295691892e-5],
+      linf = [0.0007454287896710293])
+  end
+
+  @testset "elixir_advection_amr_p4est_unstructured_flag.jl" begin
+    @test_trixi_include(joinpath(EXAMPLES_DIR, "elixir_advection_amr_p4est_unstructured_flag.jl"),
+      l2   = [1.535436496965521e-3],
+      linf = [4.400105886102593e-2])
+  end
```

Comment:
Did you compare the errors to simulations using similar resolutions and the tree/structured/unstructured meshes?

Reply:
The errors of structured conforming simulations are exactly identical to `CurvedMesh`. The errors of uncurved AMR simulations are very similar to `TreeMesh` (for example, `7.9448e-5` for `TreeMesh` vs `7.9450e-5` for `P4estMesh` when running `elixir_advection_amr_solution_independent`).